### PR TITLE
Various cleanups to ir_emitter and friends

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46686 Various cleanups to ir_emitter and friends**

I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

### CODE REVIEW

File: torch/csrc/jit/frontend/ir_emitter.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/frontend/ir_emitter.h>
+
```
Comment (SplitInfinity) [inline torch/csrc/jit/frontend/ir_emitter.cpp:2]: Intentional?

File: torch/csrc/jit/frontend/sugared_value.cpp [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/frontend/sugared_value.h>
+
```
Comment (SplitInfinity) [inline torch/csrc/jit/frontend/sugared_value.cpp:2]: Intentional?

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/python/python_sugared_value.h>
+
```
Comment (SplitInfinity) [inline torch/csrc/jit/python/python_sugared_value.cpp:2]: Intentional?

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 4]
```diff
@@ -688,28 +688,28 @@ std::shared_ptr<SugaredValue> BooleanDispatchValue::call(
...
-    error_message = inputs.at(0).value(*caller.graph());
+  } else if (args.size() == 1) {
+    error_message = args.at(0).value(*caller.graph());
   } else {
     std::vector<Value*> message_values;
-    message_values.reserve(inputs.size() + attributes.size());
+    message_values.reserve(args.size() + kwargs.size());
 
-    for (auto inp : inputs) {
+    for (auto inp : args) {
```
Comment (SplitInfinity) [inline torch/csrc/jit/python/python_sugared_value.cpp:709]:
```suggestion
    for (const auto& inp : args) {
```

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 5]
```diff
@@ -688,28 +688,28 @@ std::shared_ptr<SugaredValue> BooleanDispatchValue::call(
...
       message_values.push_back(inp.value(*caller.graph()));
     }
-    for (auto kwarg_inp : attributes) {
+    for (auto kwarg_inp : kwargs) {
```
Comment (SplitInfinity) [inline torch/csrc/jit/python/python_sugared_value.cpp:712]:
```suggestion
    for (const auto& kwarg_inp : kwargs) {
```

**Review by SplitInfinity — APPROVED** [review]
🚀

File: aten/src/ATen/core/interned_strings.h [hunk 6]
```diff
@@ -55,7 +55,7 @@ namespace c10 {
   _(prim, ReturnStmt)                \
   _(prim, BreakStmt)                 \
   _(prim, ContinueStmt)              \
-  _(prim, LocalVariableScope)        \
+  _(prim, ListComprehension)         \
```
Comment (eellison) [inline aten/src/ATen/core/interned_strings.h:58]: nit: ListComprehension -> ListComprehensionScope. it's not literally a ListComprehension, it's the scope that's created by one

**Review by eellison — COMMENTED** [review]
looks good!

File: torch/csrc/jit/python/python_sugared_value.cpp (hunk 3 above)
Comment (suo) [reply torch/csrc/jit/python/python_sugared_value.cpp:2]: yes, otherwise my clang-format reorders the includes alphabetically

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**